crates/core/src/network/networking.rs

```diff
@@ -8,34 +8,38 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use std::time::Duration;
-
-use restate_types::retries::with_jitter;
-use tracing::{info, instrument, trace};
+use tracing::{info, instrument};
 
 use restate_types::net::codec::{Targeted, WireEncode};
+use restate_types::retries::RetryPolicy;
 use restate_types::NodeId;
 
-use super::{ConnectionManager, ConnectionSender};
-use super::{NetworkError, NetworkSender};
 use crate::Metadata;
 
-const DEFAULT_MAX_CONNECT_ATTEMPTS: u32 = 10;
-// todo: make this configurable
-const SEND_RETRY_BASE_DURATION: Duration = Duration::from_millis(250);
+use super::{ConnectionManager, ConnectionSender};
+use super::{NetworkError, NetworkSender};
 
-/// Access to node-to-node networking infrastructure;
+/// Access to node-to-node networking infrastructure.
 #[derive(Clone)]
 pub struct Networking {
     connections: ConnectionManager,
     metadata: Metadata,
+    retry_policy: RetryPolicy,
+    max_retry_attempts: usize,
 }
 
 impl Networking {
-    pub fn new(metadata: Metadata) -> Self {
+    pub fn new(metadata: Metadata, retry_policy: RetryPolicy) -> Self {
+        let max_retry_attempts = match retry_policy {
+            RetryPolicy::FixedDelay { max_attempts, .. } => max_attempts,
+            RetryPolicy::Exponential { max_attempts, .. } => max_attempts,
+            _ => None,
+        };
         Self {
             connections: ConnectionManager::new(metadata.clone()),
             metadata,
+            retry_policy,
+            max_retry_attempts: max_retry_attempts.map_or(0, usize::from),
```

Comment:
`max_retry_attempts == None` means that there is no maximum after which it stops. So maybe iether `max_retry_attempts` are `Option<usize>` or we set it to `usize::MAX`.

Reply:
Maybe we shouldn't allow `RetryPolicies` with an infinite amount of attempts in the first place. Then there wouldn't be a problem.

Reply:
Agree on capping the upper bound of policies; probably out of scope for this PR but definitely a good idea.

Reply:
We should also consider adding a `max_delay` option to our policies to cap the total allowable delay due to retries.